src/Symfony/Component/Worker/Loop/Loop.php

```diff
@@ -0,0 +1,185 @@
+
+/**
+ * @author Grégoire Pineau <lyrixx@lyrixx.info>
+ */
+class Loop implements LoopInterface, ConfigurableLoopInterface
+{
+    private $router;
+    private $eventDispatcher;
+    private $logger;
+    private $name;
+    private $options;
+
+    private $stopped;
+    private $startedAt;
+    private $lastHealthCheck;
+
+    public function __construct(RouterInterface $router, EventDispatcherInterface $eventDispatcher = null, LoggerInterface $logger = null, $name = 'unnamed', array $options = array())
+    {
+        $this->router = $router;
+        $this->eventDispatcher = $eventDispatcher;
+        $this->logger = $logger;
+        $this->name = $name;
+        $this->options = array_replace(array(
+            'loop_sleep' => 200000,
+            'health_check_interval' => 10,
+        ), $options);
+
+        $this->stopped = false;
+        $this->startedAt = time();
+        // Fake the date to trigger right now a new health check.
+        $this->lastHealthCheck = 0;
+    }
+
+    public function run()
+    {
+        if (null !== $this->logger) {
+            $this->logger->notice('Worker {worker} started.', array(
+                'worker' => $this->name,
+            ));
+        }
+
+        $this->dispatch(LoopEvents::RUN);
+
+        try {
+            loop:
+
+            pcntl_signal_dispatch();
+
+            if ($this->healthCheck()) {
+                $this->dispatch(LoopEvents::HEALTH_CHECK);
+            }
+
+            if ($this->stopped) {
+                return;
+            }
+
+            $this->dispatch(LoopEvents::WAKEUP);
+
+            while (false !== $messageCollection = $this->router->fetchMessage()) {
+                if (!$messageCollection instanceof MessageCollection) {
+                    throw new \RuntimeException('This is not a MessageCollection instance.');
+                }
+                if (null !== $this->logger) {
+                    $this->logger->notice('New message.');
+                }
+
+                $result = $this->router->consume($messageCollection);
+
+                if (null !== $this->logger) {
+                    if (false === $result) {
+                        $this->logger->warning('Messages consumed with failure.');
+                    } else {
+                        $this->logger->info('Messages consumed successfully.');
+                    }
+                }
+
+                pcntl_signal_dispatch();
+
+                if ($this->healthCheck()) {
+                    $this->dispatch(LoopEvents::HEALTH_CHECK);
+                }
+
+                if ($this->stopped) {
+                    return;
+                }
+            }
+
+            $this->dispatch(LoopEvents::SLEEP);
+
+            usleep($this->options['loop_sleep']);
+
+            goto loop;
+        } catch (StopException $e) {
+            $this->stop('Force shut down of the worker because a StopException has been thrown.', $e);
+
+            return;
+        } catch (\Exception $e) {
```

Comment:
Are you referring to the fact we do nothing in the catch block?

Indeed it can be hard to understand. Initially we had 2 catch block (AMQPException and Exception) with 2 different log messages. So it's a relic of our previous lib.

Now it could be simplified indeed. Except It guess I have to support FatalError too. ;)